sklearn/plot/_gridsearch_results.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+from sklearn.plot import plot_heatmap
+
+
+def plot_gridsearch_results(cv_results, metric='mean_test_score',
+                            xlabel=None, ylabel=None,
+                            title='Grid Search Results', cmap=None,
+                            vmin=None, vmax=None, ax=None, fmt="{:.2f}",
+                            xtickrotation=45, norm=None):
+    """Print and plot the confusion matrix as a heatmap. Normalization can be
+    applied by setting `normalize=True`.
+
+    Parameters
+    ----------
+    cv_results : dict of numpy (masked) ndarrays
+        The cv_results_ attribute of the GridSearchCV object.
+
+    xlabel : string, default=None
+        Label for the x-axis. If None, the first key of the param_grid will
+        be used as the xlabel.
+
+    ylabel : string, default=None
+        Label for the y-axis. If None, the second key of the param_grid will
+        be used as the ylabel.
+
+    metric : string, default="mean_test_score"
+        The metric from the GridSearchCV results to display. This is ignored
+        if only 1 parameter is used in grid search.
+
+    title : string, default="Grid Search Results"
+        Title for the heatmap.
+
+    cmap : string or colormap
+        Matpotlib colormap to use.
+
+    vmin : int, float or None
+        Minimum clipping value. This argument will be passed on to the
+        pcolormesh function from matplotlib used to generate the heatmap.
+
+    vmax : int, float or None
+        Maximum clipping value. This argument will be passed on to the
+        pcolormesh function from matplotlib used to generate the heatmap.
+
+    ax : axes object or None
+        Matplotlib axes object to plot into. If None, the current axes are
+        used.
+
+    fmt : string, default="{:.2f}"
+        Format string to convert value to text.
+
+    xtickrotation : float, default=45
+        Rotation of the xticklabels.
+
+    norm : matplotlib normalizer
+        Normalizer passed to pcolormesh function from matplotlib used to
+        generate the heatmap.
+    """
+
+    import matplotlib.pyplot as plt
+
+    params = sorted(cv_results['params'][0].keys())
+    nparams = len(params)
+
+    if nparams == 1:
```

Comment:
because the same steps are being repeated in all the 3 functions. if we have 1 function with 3 cases, then the fixed steps are ran once then the parts specific to each case are executed.